### ISSUE TITLE
Opening non-ext4 system panics

### ISSUE THREAD

**Issue by Mathis-Z** [nicholasbishop/ext4-view-rs#280]
I wanted to use this crate to test if a block device is an ext4 system. However, when opening a non-ext4 system it panicked:

```
panicked at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14/library/core/src/num/mod.rs:1136:5:
attempt to multiply with overflow
```

Before the panic it only read the first 1024 bytes from the block device. These are the hex-encoded bytes that were read:

```
8955e4894dd431c03b45d87d0ec7048500031000ffffffff40ebed31c03b45e47d0a8b7dd4c604070040ebf18b45e4408945dc31c9c745e0000000008b5de0395dd80f8ecb00000031db395de47e0d80bc0b0010100000750d43ebee7508ff45e0034ddcebd68b45e0891c85000310000fb6840b001010000fb69000021000b8ff00000029d031f68db900101000897dd00fb690000010003975e47c1f894dc88b45d08d3c300fb6078955cce81cffffff8807468b4dc88b55ccebdc8b45e0408945d08b45dc8d3c018b75d03975d8748d31f60fb6943b001010008d87001010008945cc3975e47c220fb6840e00101000894dc48955c8e8d1feffff8b55cc300432468b4dc48b55c8ebd9ff45d0037ddcebb68b75d84e8b45e4f7d08945d88b5ddc0fafde035de485f678578b3cb50003100083ffff744589d82b45e48945dcc645e00031c9394de47e238b45d40fb614088b45dc0fb6840100101000894dd0e868feffff3045e08b4dd041ebd88a45e03283001010008b4dd48804394e035dd8eba583c43c5b5e5f5dc35589e557565383ec4c8945bc8955dc894de085c90f84fb02000001d08945d48d4c08ff89c83b45d47208803800750348ebf329c1bb020000008b45e099f7fb39c10f8fce020000c6050002100000ba00001000b00188028882ff0000000fb6c888910002100084c079088d040983f01deb02d1e04281faff00100075d8837ddc000f848e0200008b4ddc034de081f90090010076246945dcc800000031d2f7f1c1e0098945d06945e0c800000031d2f7f1c1e0098945cceb0c8b45e08945cc8b45dc8945d0c745c0ff010000c745c8000000008b45d00345c08b55cc0355c081faff010000767f3dff0100007678c1e8098945d88b45bc0345c88945c489c131c0e98a00000089d7c1ef0a31db8b45d88b75e48d7430ff89d989f2b8000e1000e8dbfcffff8883000d100043395de47fe631c080b8000d1000000f8594000000403945e47fed0f858800000031c08a90000e10008b7dc48817408145c4000200003945d87fe8ff45c8ff4dc0817dc8000200000f855affffff8b5dd0015dbc8b7dcc017dd4295ddc297de0e9edfeffff3b45d87d118a198898000e10004081c100020000ebea89d0c1e8098945e48b4dd4034dc831c03b45e40f8d47ffffff8a198b75d8889c06000e10004081c100020000ebe28d770131c931d2eb168a9c02000d1000889c08001010004039f87eed4201f131c039fa7ce439c77e0ac680000710000040ebf2b90007100089fa89f8e854fcffff31f631db8b45d80345e48945b8488945b0895db43b75b87d35b9ff00000029f18d57ffb800071000e8befbffff3a86000010007516888300081000438b45b029f08b55b48904950009100046ebc331c039c3740ac680001010000140ebf2a000
```

Happy bug-hunting :sweat_smile: 

Edit: I forgot to pad the bytes to two hex characters. I corrected that.

**Comment by nicholasbishop** [comment]
Thanks for the bug report. I've got a [PR](https://github.com/nicholasbishop/ext4-view-rs/pull/284) up to fix it, and also filed https://github.com/nicholasbishop/ext4-view-rs/issues/283 to help catch similar issues.

**Comment by nicholasbishop** [comment]
Released 0.6.1 with this fix.
